core_types/src/link/mod.rs

```diff
@@ -63,6 +64,27 @@ pub enum LinkActionKind {
     REMOVE,
 }
 
+pub enum LinkMatch<S: Into<String>> {
+    Any,
+    Exactly(S),
+    Regex(S),
+}
+
+impl<S: Into<String>> LinkMatch<S> {
+    pub fn to_regex_string(self) -> Result<String, String> {
+        let re_string: String = match self {
+            LinkMatch::Any => ".*".into(),
+            LinkMatch::Exactly(s) => "^".to_owned() + &s.into() + "$",
```

Comment:
Thanks @pdaoust I didn't consider that tags could be weird regexes when the caller has no intention of them being treated as such.

On the topic of accepting `Into<String>` this doesn't work well in the case of passing an `Option<String>` (or in this case a LinkMatch<String>) because it cannot infer the type for the None/Any variant which doesn't accept a parameter! You then need to annotate the type (e.g. `LinkMatch<String>::Any`) so I am not sure if it actually makes it any simpler overall. The errors are more cryptic in this case as well.

I made the compromise of having get_links accept a LinkMatch<&str> since I expect for the most part it will be called with static strings. It is just as simple to convert a String type to a slice when calling it with a String.